### ISSUE TITLE
mov dr8, r9

### ISSUE THREAD

**Issue by mrexodia** [keystone-engine/keystone#105]
```
0x00007FF6C0B313B0 "mov dr8, r9" failed to assemble (ks_asm() failed: count = 0, error = 512):
    Expected (04) 4F 0F 23 01 
```

**Comment by Nukem9** [comment]
Am I missing something or is `dr8` not a valid register, making this a non-issue...?

**Comment by mrexodia** [comment]
I don't know, I just disassembled 20mb of data and checked where capstone didn't match keystone.

**Comment by mrexodia** [comment]
So it might very well be a capstone bug

**Comment by aquynh** [comment]
That might be the case. Can somebody write a simple test in kernel (like Windows driver) to confirm this is a valid instruction?

**Comment by Nukem9** [comment]
```
Intel® 64 and IA-32 Architectures
Software Developer’s Manual

In 64-bit mode, more encodings for control and debug registers are available. The REX.R bit is used to modify the
ModR/M reg field when that field encodes a control or debug register (see Table 2-4). These encodings enable the
processor to address CR8-CR15 and DR8-DR15. An additional control register (CR8) is defined in 64-bit mode. CR8
becomes the Task Priority Register (TPR).

In the first implementation of IA-32e mode, CR9-CR15 and DR8-DR15 are not implemented. Any attempt to access
unimplemented registers results in an invalid-opcode exception (#UD).
```

tl;dr
CR0, CR2, CR3, CR4, or CR8 are **valid/implemented**
DR0, DR1, DR2, DR3, DR4, DR5, DR6, or DR7 are **valid/implemented**

CR9-CR15 and DR8-DR15 are **valid** instructions, but **unimplemented** and will raise #UD when executed.

**Comment by aquynh** [comment]
Yes, but what i mean is to actually verify the encoding because this can be undocumented instruction. Intel manual is known untrusted.
